content/blog/nats-server-2.9.15-release.md

```diff
@@ -0,0 +1,132 @@
+title = "NATS Server 2.9.15 Release"
+author = "Byron Ruth"
+categories = ["Engineering"]
+tags = ["NATS", "Testing", "Quality"]
++++
+
+Normally, the NATS maintainers do not publish a dedicated announcement for a patch release, but the [2.9.15 release](https://github.com/nats-io/nats-server/releases/tag/v2.9.15) of the NATS server has several significant stability and performance improvements worth highlighting, all driven by our new quality engineering process.
+
+## Release management
+
+What does a patch release mean to NATS? And what are these new quality engineering processes?
+
+At the beginning of 2023, the NATS maintainers drafted an end-to-end “release management” document which includes the policies and procedures used for each NATS server release, everything from initial scoping to communication of the release.
+
+This effort was driven largely by acknowledging the ever-increasing number of critical, high-scale production NATS deployments companies rely on.
+
+For how basic it may sound, one such policy is adherence to the SemVer semantics and explicitly defining what a patch, minor, and major release means for NATS.
+
+Concretely, a NATS server patch release is allowed to have the following types of changes assuming they maintain functional compatibility with all existing clients.
+
+- Bug fixes
+- Security patches
+- Performance improvements
+- Code documentation
+
+## Quality engineering
+
+Quality engineering encompasses the spectrum of burn-in testing and quality assurance practices. Most notably, the testing process has evolved significantly over the past couple months to introduce complex topologies, fault injection, and stress testing.
+
+To reiterate, this release is significant as it is the first patch release where many of these new techniques have been applied.
+
+## Get on with it
+
+So what are all of these great changes??
+
+Historically we would point you to the [release notes on GitHub](https://github.com/nats-io/nats-server/releases/tag/v2.9.15), but going forward, as part of improving our “release communication,” we will focus on the observed behavior first and link them to the relevant code changes.
+
+## 👻 Ghost consumers
+
+There were a handful of inbound issues where consumers were being "cleaned up" without expectation or active intention. Cases included:
+
+- Durable consumers being deleted prior to an activity threshold being reached
+- Ephemeral consumers deleted while there was still an active subscription
+- Durable consumers being deleted on hard restart even when they are recoverable
+
+The two root causes were over-active timers for tracking activity thresholds and improperly propagating errors to the JetStream "meta-layer" which handles the lifecycle of streams and consumers.
+
+The *ghost factor* comes from the fact that not all replicas would report the existence or deletion of a given consumer. NATS CLI consumer reports may list a consumer, but the consumer info would be fail to report.
```

Comment:
"but the consumer info would be fail to report" => should be "would fail to report" ?